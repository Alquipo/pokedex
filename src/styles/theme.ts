export default {
  grid: {
    container: '140rem',
    gutter: '3.2rem'
  },

  border: {
    radius: {
      xsmall: '0.3rem',
      small: '0.5rem',
      large: '1.0rem'
    }
  },

  font: {
    family:
      "Poppins, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    medium: 600,
    normal: 400,
    bold: 700,
    sizes: {
      xsmall: '1.2rem',
      small: '1.4rem',
      medium: '1.6rem',
      large: '1.8rem',
      xlarge: '2.0rem',
      xxlarge: '3.5rem'
    }
  },

  colors: {
    // primary: '#ecf0f1',
    // secondary: '#e4c439',
    // mainBg: '#06092B',
    // black: '#030517',
    // darkGray: '#2E2F42',
    lightBg: '#f2f3f5',
    white: '#fff',
    lightGray: '#EAEAEA',
    gray: '#505050',

    pokemonType: {
      normal: '#A0A29F',
      fire: '#FBA54C',
      water: '#539DDF',
      electric: '#F2D94E',
      grass: '#5FBD58',
      ice: '#75D0C1',
      ground: '#DA7C4D',
      flying: '#A1BBEC',
      ghost: '#5F6DBC',
      rock: '#C9BB8A',
      fighting: '#D3425F',
      poison: '#B763CF',
      psychic: '#FA8581',
      bug: '#92BC2C',
      dark: '#595761',
      steel: '#5695A3',
      dragon: '#0C69C8',
      fairy: '#EE90E6'
    },

    pokemonText: {
      normal: '#bcaaa4',
      fire: '#ff8a80',
      water: '#81d4fa',
      electric: '#ffd600',
      grass: '#81c784',
      poison: '#ad8ee7',
      fairy: '#f8bbd0',

      bug: '#81c784',
      ice: '#81d4fa',
      ground: '#bcaaa4',
      flying: '#81d4fa',
      ghost: '#ad8ee7',
      rock: '#bcaaa4',
      fighting: '#ff8a80',
      psychic: '#ff8a80',
      dark: '#bcaaa4',
      steel: '#5695A3',
      dragon: '#81d4fa',

      transparent: 'hsla(0,0%,100%,0.5)'
    }
  },

  transition: {
    color: 'color 1000ms ease-in, width 1500ms ease-in-out',
    backgroundColor: 'background-color 1000ms ease-in, width 1500ms ease-in-out'
  },

  spacings: {
    xxsmall: '0.3rem',
    xsmall: '0.8rem',
    small: '1.6rem',
    medium: '3.2rem',
    large: '4.0rem',
    xlarge: '4.8rem',
    xxlarge: '7.5rem'
  },

  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50
  }
} as const
