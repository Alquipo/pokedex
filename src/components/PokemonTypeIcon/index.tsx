import Image from 'next/image'

import { PokemonTypesProps } from 'components/PokemonCard'
import * as S from './styles'

export type PokemonTypeIconProps = {
  type: PokemonTypesProps
  hasHover?: boolean
}

const PokemonTypeIcon = ({ type, hasHover = false }: PokemonTypeIconProps) => (
  <S.Wrapper typeIcon={type} hasHover={hasHover}>
    <Image
      src={`/img/type-icons/${type}.png`}
      alt={type}
      width={22}
      height={22}
    />
  </S.Wrapper>
)

export default PokemonTypeIcon
