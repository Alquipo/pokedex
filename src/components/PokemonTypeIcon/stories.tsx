import { Story, Meta } from '@storybook/react/types-6-0'
import PokemonTypeIcon, { PokemonTypeIconProps } from '.'

export default {
  title: 'Reusable/PokemonTypeIcon',
  component: PokemonTypeIcon,
  args: {
    type: 'fire'
  }
} as Meta

export const Default: Story<PokemonTypeIconProps> = (args) => (
  <div
    style={{
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'flex-end',

      height: '16rem',
      width: '18rem ',

      position: 'relative',

      padding: '0.3rem',
      margin: '0.3rem 0.8rem',

      borderRadius: '1rem',
      border: '0.1rem solid',
      textAlign: 'center',

      cursor: 'default',

      backgroundColor: 'white'
    }}
  >
    <PokemonTypeIcon {...args} />
    <PokemonTypeIcon {...args} />
  </div>
)
