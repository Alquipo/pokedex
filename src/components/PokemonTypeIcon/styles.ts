import styled, { css } from 'styled-components'
import { PokemonTypesProps } from 'components/PokemonCard'
import { PokemonTypeIconProps } from '.'
import media from 'styled-media-query'

type WrapperProps = {
  typeIcon: PokemonTypesProps
} & Pick<PokemonTypeIconProps, 'hasHover'>

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, typeIcon, hasHover }) => css`
    border-radius: 100%;
    height: 3rem;
    width: 3rem;
    transition: all 0.2s;
    margin: 2px 3px;

    display: flex;
    align-items: center;
    justify-content: center;

    background-color: ${theme.colors.pokemonType[typeIcon]};
    box-shadow: 0 0 2rem ${theme.colors.pokemonType[typeIcon]};

    ${media.greaterThan('medium')`
      ${
        hasHover &&
        `&:hover {
        filter: saturate(200%);
        transform: scale(1.1);
      }`
      }}
    `}
  `}
`
