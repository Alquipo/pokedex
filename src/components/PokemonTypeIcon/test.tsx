import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'
import theme from 'styles/theme'

import PokemonTypeIcon, { PokemonTypeIconProps } from '.'

const props: PokemonTypeIconProps = {
  type: 'fire'
}

jest.mock('next/image', () => ({
  __esModule: true,
  default: function Mock({ children }: { children: React.ReactNode }) {
    return (
      <img
        data-testid="Mock Image"
        alt={props.type}
        src={`/img/type-icons/${props.type}.png`}
      >
        {children}
      </img>
    )
  }
}))

describe('<PokemonTypeIcon />', () => {
  it('should render the PokemonTypeIcon correctly', () => {
    const { container } = renderWithTheme(<PokemonTypeIcon {...props} />)

    expect(screen.getByTestId('Mock Image')).toBeInTheDocument()

    expect(screen.getByRole('img', { name: props.type })).toHaveAttribute(
      'src',
      `/img/type-icons/fire.png`
    )

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the style PokemonTypeIcon', () => {
    const { container } = renderWithTheme(<PokemonTypeIcon {...props} />)

    expect(container.firstChild).toHaveStyle({
      backgroundColor: theme.colors.pokemonType[props.type],
      boxShadow: `0 0 2rem ${theme.colors.pokemonType[props.type]}`
    })
  })
})
