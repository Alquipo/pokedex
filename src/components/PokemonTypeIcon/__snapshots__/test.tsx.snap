// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonTypeIcon /> should render the PokemonTypeIcon correctly 1`] = `
.c0 {
  border-radius: 100%;
  height: 3rem;
  width: 3rem;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  margin: 2px 3px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #FBA54C;
  box-shadow: 0 0 2rem #FBA54C;
}

@media (min-width:768px) {

}

<div
  class="c0"
>
  <img
    alt="fire"
    data-testid="Mock Image"
    src="/img/type-icons/fire.png"
  />
</div>
`
